libportal/remote.c

```diff
@@ -746,7 +887,7 @@ xdp_session_start (XdpSession *session,
 
   call = g_new0 (StartCall, 1);
   call->portal = g_object_ref (_xdp_session_get_portal (session));
-  call->session = g_object_ref (session);
+  call->session = g_object_ref (XDP_SCREENCAST_SESSION (session));
```

Comment:
Do you really need to cast here? `g_object_ref()` returns `gpointer` which shouldn't trigger any warnings